### PR TITLE
User meeting routes

### PR DESCRIPTION
**Pull request by inathan44**

### CODE REVIEW

File: server/api/routes/userMeeting.cjs [hunk 1]
```diff
@@ -1,33 +1,107 @@
...
+  async (req, res, next) => {
+    /**
+     * GET /api/user/:userId/meeting/:meetingId
+     * get details of a single meeting
+     */
+    try {
+      const meeting = await Meeting.findByPk(req.params.meetingId);
+      if (!meeting) {
+        return res.status(404).send('No meeting found.');
+      } else if (req.user.id !== meeting.userId && req.user.role !== 'admin') {
```
Comment (pbryan9) [inline server/api/routes/userMeeting.cjs:36]: req.user.id can be either meeting.userId OR meeting.buddyId

File: server/api/routes/userMeeting.cjs [hunk 2]
```diff
@@ -1,33 +1,107 @@
...
-router.get('/', (req, res, next) => {
+router.get('/', requireToken, sameUserOrAdmin, async (req, res, next) => {
   /**
    * GET /api/user/:userId/meeting
    * return a list of all meetings associated with userId
    */
-
-  res.send(`hello from /api/user/${req.params.userId}/meeting`);
+  try {
+    const userMeetings = await Meeting.findAll({
```
Comment (pbryan9) [inline server/api/routes/userMeeting.cjs:11]: (just taking the note from your own catch - well done) -- need to check userId for existence (User.findByPk(:userId))

File: server/api/routes/userMeeting.cjs [hunk 3]
```diff
@@ -1,33 +1,107 @@
...
+  async (req, res, next) => {
+    /**
+     * PUT /api/user/:userId/meeting/:meetingId/cancel
+     * set meeting status to 'closed'
+     */
+    try {
+      const meeting = await Meeting.findByPk(req.params.meetingId);
+      if (!meeting) {
+        return res.status(404).send('No meeting found to update.');
+      } else if (req.user.id !== meeting.userId && req.user.role !== 'admin') {
```
Comment (pbryan9) [inline server/api/routes/userMeeting.cjs:92]: need buddy access here as well

File: server/api/routes/userMeeting.cjs [hunk 4]
```diff
@@ -1,33 +1,107 @@
...
+        res.status(403).send('You are unable to edit this meeting.');
+      } else {
+        const updatedMeeting = await meeting.update({ isClosed: true });
+        res.status(200).json(updatedMeeting);
+      }
+    } catch (error) {
+      console.error(
+        'Error trying to change meeting date. Please try again later.'
```
Comment (pbryan9) [inline server/api/routes/userMeeting.cjs:100]: close*

**Comment by inathan44** [comment]
Added a check to the routes that have userId and meetingId in the route to make sure that the meeting that was being queried belonged to that user. Before this check, an admin could change a meeting (for ex meeting 15) even though they passed in a user is that DID NOT belong to that meeting.

A new middleware is added in to check if the user in params is a valid user.

queries now pull meetings where user is a buddy and allows the buddy to update meetings as well.

console.error bug was also fixed

File: server/api/routes/userMeeting.cjs [hunk 5]
```diff
@@ -1,33 +1,161 @@
...
+          id: req.params.meetingId,
+        },
+      });
+      if (!meeting) {
+        return res
+          .status(404)
+          .send('No meeting found matching user and meetingId to update.');
+      } else if (
+        req.user.id !== meeting.userId &&
+        req.user.id !== meeting.buddyId &&
```
Comment (inathan44) [inline server/api/routes/userMeeting.cjs:103]: Remove buddy ability to change the time of the meeting

**Review by pbryan9 — APPROVED** [review]

**Pull request merged**